.circleci/config.yml

```diff
@@ -20,36 +20,52 @@ jobs:
           name: Write Job name
           command: echo $CIRCLE_JOB > circlejob.txt
 
-      - restore_cache:
-          key: deps1-{{ .Branch }}-{{ checksum "dev-requirements.txt" }}-{{ checksum "package.json" }}-{{ checksum ".circleci/config.yml" }}-{{ checksum "circlejob.txt" }}
-
       - run:
           name: Install dependencies
           command: |
-              sudo pip install virtualenv --upgrade
+              sudo pip install --upgrade virtualenv
               python -m venv venv || virtualenv venv
               . venv/bin/activate
-              pip install -e .
-              pip install -r dev-requirements.txt --quiet
-              pip list | grep dash
+              pip install -r requirements-base.txt --quiet
               npm install --ignore-scripts
 
-      - run:
-          name: Build
-          command: npm run build
-
-      - save_cache:
-          key: deps1-{{ .Branch }}-{{ checksum "dev-requirements.txt" }}-{{ checksum "package.json" }}-{{ checksum ".circleci/config.yml" }}-{{ checksum "circlejob.txt" }}
-          paths:
-              - "venv"
-              - "node_modules"
-
       - run:
           name: Lint & format
           command: |
               npm run lint:test
               npm run format:test
 
+      - run:
+          name: Install dependencies (transient)
+          command: |
+              . venv/bin/activate
+              git clone git@github.com:plotly/dash.git
+              pip install -e ./dash --quiet
+              git clone git@github.com:plotly/dash-core-components.git
+              cd dash-core-components && npm i && cd ..
+              cd dash-core-components && npm run build && cd ..
+              pip uninstall dash-core-components -y
+              pip install -e ./dash-core-components --quiet
+              git clone git@github.com:plotly/dash-html-components.git
+              cd dash-html-components && npm i && cd ..
+              cd dash-html-components && npm run build && cd ..
+              pip uninstall dash-html-components -y
+              pip install -e ./dash-html-components --quiet
+              git clone git@github.com:plotly/dash-table.git
+              cd dash-table && npm i && cd ..
+              cd dash-table && npm run build && cd ..
+              pip uninstall dash-table -y
+              pip install -e ./dash-table --quiet
```

Comment:
renderer tests were already using html but not asking for them explicitly -- fetch everything and ask questions later..

Reply:
all of this is very verbose right now -- want to wait until the same logic is applied in all of dash-core before refactoring the 🍝

Reply:
thinking of a script that takes the repo's name and the branch name in the dash-core repo atm

Reply:
the explicit build also brings us a step closer to removing the builds from the repo if desired

Reply:
Yeah, I think this scheme makes sense. It's certainly easier conceptually - and more flexible, as we have the full dev version of the repo - than trying to do this all via `pip` machinery. It might be nice at some point to do it in a cacheable way - something like check if each repo exists locally; if it does just do a `git pull/checkout` on the appropriate branch; if not, do a `git clone`.

Can you also group all the branch references together, so it's easy to see which ones are on master and which aren't? I guess if there's a script that we just call multiple times, that would take care of this... if not, we could just make vars for the branch names.

Is the `pip uninstall` step really needed if you're just going to do `install -e` right after?

Reply:
Yes, I can group things together. I was thinking of scripting this with repo+branch as params later on. I'm not sure about uninstall vs. -e, might just be a bit paranoid again, will test it out.

I can also make master explicit by always using `-b master` even if it's redundant.

Reply:
After the grouping & simplifying this looks great, probably not even worth adding a script as it'll be easier to follow inline 🏆